pkg/kubectl/cmd/secrets/dockercfg.go

```diff
@@ -0,0 +1,169 @@
+	EmailAddress     string
+
+	SecretsInterface client.SecretsInterface
+
+	Out io.Writer
+}
+
+// NewCmdGet creates a command object for the generic "get" action, which
+// retrieves one or more resources from a server.
+func NewCmdCreateDockerConfigSecrets(name, fullName string, f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	o := &CreateDockerConfigOptions{Out: out}
+
+	cmd := &cobra.Command{
+		Use:   name + " <secret name>",
+		Short: "Create a new dockercfg secret",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := o.Complete(f, args); err != nil {
+				cmdutil.CheckErr(err)
+			}
+
+			if err := o.Validate(); err != nil {
+				cmdutil.CheckErr(err)
+			}
+
+			if err := o.CreateDockerSecret(); err != nil {
+				cmdutil.CheckErr(err)
+			}
+
+		},
+	}
+
+	cmd.Flags().StringVar(&o.Username, "docker-username", "", "username for docker registry authentication")
+	cmd.Flags().StringVar(&o.Password, "docker-password", "", "password for docker registry authentication")
+	cmd.Flags().StringVar(&o.EmailAddress, "docker-email", "", "email for docker registry")
+	cmd.Flags().StringVar(&o.RegistryLocation, "docker-server", "https://index.docker.io/v1/", "server location for docker registry")
+
+	return cmd
+}
+
+func (o CreateDockerConfigOptions) CreateDockerSecret() error {
+	if err := o.Validate(); err != nil {
+		return err
+	}
+
+	dockercfgAuth := credentialprovider.DockerConfigEntry{
+		Username: o.Username,
+		Password: o.Password,
+		Email:    o.EmailAddress,
+	}.ConvertToDockerConfigCompatible()
+
+	dockerCfg := map[string]credentialprovider.DockerConfigEntryWithAuth{o.RegistryLocation: dockercfgAuth}
+
+	dockercfgContent, err := json.Marshal(dockerCfg)
+	if err != nil {
+		return err
+	}
+
+	secret := &api.Secret{}
+	secret.Namespace = o.SecretNamespace
+	secret.Name = o.SecretName
+	secret.Type = api.SecretTypeDockerConfig
+	secret.Data = map[string][]byte{}
+	secret.Data[api.DockerConfigKey] = dockercfgContent
+
+	if _, err := o.SecretsInterface.Create(secret); err != nil {
+		return err
+	}
+
+	fmt.Fprintf(o.GetOut(), "%s/%s\n", secret.Namespace, secret.Name)
+
+	return nil
+}
+
+func (o *CreateDockerConfigOptions) Complete(f *cmdutil.Factory, args []string) error {
+	if len(args) != 1 {
+		return errors.New("must have exactly one argument: secret name")
+	}
+	o.SecretName = args[0]
+
+	client, err := f.Client()
+	if err != nil {
+		return err
+	}
+	o.SecretNamespace, err = f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	o.SecretsInterface = client.Secrets(o.SecretNamespace)
+
+	return nil
+}
+
+func (o CreateDockerConfigOptions) Validate() error {
+	if len(o.SecretNamespace) == 0 {
+		return errors.New("SecretNamespace must be present")
+	}
+	if len(o.SecretName) == 0 {
+		return errors.New("secret name must be present")
+	}
+	if len(o.RegistryLocation) == 0 {
+		return errors.New("docker-server must be present")
+	}
+	if len(o.Username) == 0 {
+		return errors.New("docker-username must be present")
+	}
```

Comment:
should we prevent usernames that contain `:`, since they can't be encoded into a basic-auth header?